ts/pages/staking/wizard/wizard.tsx

```diff
@@ -252,34 +254,13 @@ const SetupStaking: React.FC<SetupStakingProps> = ({
     poolId,
     onGoToNextStep,
     onUpdateAccountBalances,
+    account,
+    connector,
 }) => {
-    let wizardFlowStep: WizardSetupSteps = WizardSetupSteps.Empty;
-    if (providerState.account.state !== AccountState.Ready) {
-        wizardFlowStep = WizardSetupSteps.ConnectWallet;
-    } else if (!zrxBalanceBaseUnitAmount) {
-        // TODO(johnrjj) - This should also add a spinner to the ConnectWallet panel
-        // e.g. at this point, the wallet is connected but loading stuff via redux/etc
-        wizardFlowStep = WizardSetupSteps.ConnectWallet;
-    } else if (!zrxBalance || zrxBalance.isLessThan(1)) {
-        // No balance...
-        wizardFlowStep = WizardSetupSteps.NoZrxInWallet;
-    } else {
-        // Otherwise, we're good to show either the MM or Recommendation pane
-        if (poolId) {
-            // Coming from market maker entry
-            wizardFlowStep = WizardSetupSteps.MarketMakerEntry;
-        } else {
-            // Coming from wizard/recommendation entry
-            wizardFlowStep = WizardSetupSteps.RecomendedEntry;
-        }
-    }
```

Comment:
Can we keep this logic up here?

I know it's a tempting refactor to make it cleaner, but I prefer the if statements and enums, as they are more explicit about what's going on in the wizard logic to someone reading the code

Reply:
Personally, would love to but it breaks the changes. I was sad removing it :joy: . The account shows up as undefined at all time even when the user is logged in

Reply:
How about we remove the undefined account part then

```ts
    let wizardFlowStep: WizardSetupSteps = WizardSetupSteps.Empty;
    if (!zrxBalanceBaseUnitAmount) {
        // TODO(johnrjj) - This should also add a spinner to the ConnectWallet panel
        // e.g. at this point, the wallet is connected but loading stuff via redux/etc
        wizardFlowStep = WizardSetupSteps.ConnectWallet;
    } else if (!zrxBalance || zrxBalance.isLessThan(1)) {
        // No balance...
        wizardFlowStep = WizardSetupSteps.NoZrxInWallet;
    } else {
        // Otherwise, we're good to show either the MM or Recommendation pane
        if (poolId) {
            // Coming from market maker entry
            wizardFlowStep = WizardSetupSteps.MarketMakerEntry;
        } else {
            // Coming from wizard/recommendation entry
            wizardFlowStep = WizardSetupSteps.RecomendedEntry;
        }
    }
```

Reply:
or port to the new web3-react account, that would probably work

```

const { account } = useWeb3React();

// ...

let wizardFlowStep: WizardSetupSteps = WizardSetupSteps.Empty;
    if (account === undefined) {
        wizardFlowStep = WizardSetupSteps.ConnectWallet;
    } else if (!zrxBalanceBaseUnitAmount) {
        // TODO(johnrjj) - This should also add a spinner to the ConnectWallet panel
        // e.g. at this point, the wallet is connected but loading stuff via redux/etc
        wizardFlowStep = WizardSetupSteps.ConnectWallet;
    } else if (!zrxBalance || zrxBalance.isLessThan(1)) {
        // No balance...
        wizardFlowStep = WizardSetupSteps.NoZrxInWallet;
    } else {
        // Otherwise, we're good to show either the MM or Recommendation pane
        if (poolId) {
            // Coming from market maker entry
            wizardFlowStep = WizardSetupSteps.MarketMakerEntry;
        } else {
            // Coming from wizard/recommendation entry
            wizardFlowStep = WizardSetupSteps.RecomendedEntry;
        }
    }
```

Reply:
The issue here is that the data is outside the render method will be called once and it gets called before the page is completely loaded

Reply:
Both failed at all times